widgets/comment/Comment.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace app\widgets\comment;
+
+use app\components\object\ObjectIdentityInterface;
+use app\models\Comment as CommentForm;
+use Yii;
+use yii\base\InvalidConfigException;
+use yii\base\Widget;
+use yii\db\ActiveRecord;
+
+/**
+ * Comment widget
+ */
+class Comment extends Widget
+{
+    /**
+     * @var ActiveRecord|ObjectIdentityInterface
+     */
+    public $model;
+
+    /**
+     * @throws InvalidConfigException
+     */
+    public function init()
+    {
+        if ($this->model === null) {
+            throw new InvalidConfigException('Comment widget property model is not set.');
+        }
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function run()
+    {
+        $commentForm = null;
+        if (!Yii::$app->user->isGuest) {
+            $commentForm = new CommentForm([
+                'scenario' => CommentForm::SCENARIO_CREATE,
+                'object_type' => $this->model->getObjectType(),
+                'object_id' => $this->model->getObjectId()
+            ]);
+            $commentForm->loadDefaultValues();
+
+            if ($commentForm->load(Yii::$app->request->post()) && $commentForm->save()) {
+                Yii::$app->getResponse()->refresh(sprintf("#c%d", $commentForm->id))->send();
```

Comment:
Could return here instead of using `Yii::$app->end()`.

Reply:
Now I looked, maybe `return` is not needed.

```php
Yii::$app->getResponse()->refresh(sprintf("#c%d", $commentForm->id))->send();
```

`send()` - itself closes the application.

PS: it's a widget, not a controller.

Reply:
`return` is to ensure no code from the rest of the method is executed.

Reply:
it is yes) Fixed.